pkg/installmanager/installmanager.go

```diff
@@ -1194,17 +1197,21 @@ func (m *InstallManager) setupSSHKnownHost(knownHost string) error {
 		m.log.WithError(err).Error("error writing to /etc/passwd")
 		return err
 	}
+	return nil
+}
 
-	sshDir := "/home/hive/.ssh"
+func (m *InstallManager) writeSSHKnownHosts(homeDir string, knownHosts []string) error {
+	sshDir := filepath.Join(homeDir, ".ssh")
 	if err := os.MkdirAll(sshDir, 0700); err != nil {
 		m.log.WithError(err).Errorf("error creating %s directory", sshDir)
 		return err
 	}
-	if err := ioutil.WriteFile(filepath.Join(sshDir, "known_hosts"), []byte(knownHost), 0644); err != nil {
+	allKnownHosts := strings.Join(knownHosts[:], "\n")
+	if err := ioutil.WriteFile(filepath.Join(sshDir, "known_hosts"), []byte(allKnownHosts), 0644); err != nil {
```

Comment:
Is there any chance this file would ever exist and contain content that needs to be preserved?

Reply:
+1. See https://github.com/openshift/hive/pull/723#discussion_r356062465.

Reply:
If it's in the container image something is really wrong. If it's mounted in via kube we're going to fail to be able to write it. (IIRC) I don't really want to take on the edge cases around newlines at the end of file or not for something that really shouldn't be a concern. 

How about a hard error if the file exists before we go to write and we can deal with it if that comes up.

Reply:
OR we're using /output as HOME and re-trying a failed install. :) But in this case we still want to overwrite rather than append or check if we need to append, which would lean me towards keeping it as is. Thoughts?

Reply:
I don't understand the second scenario. However, I am fine with either a hard error or a warning log entry if the file exists.

Reply:
The only reasons for the file to exist that I could think of are the existing volume and a custom image. In the case of the volume, it seems like you're right and we should consider that a re-try attempt and write over the file. So, do we support custom images for any reason?

Reply:
Custom images, only for devs testing changes really.

Reply:
I don't think we need to spend a lot of energy on custom developer images, so if we don't support them for users I think it's safe to always try to overwrite the file.

Reply:
Yeah, the more I think about this, the more that I am OK with doing nothing here. We don't need to do anything to detect an existing file.

Reply:
Just don't ever try to run this binary outside of a container. ;)

Reply:
:boom:

Reply:
That's what backups are for. :)

Thanks, feedback addressed then.